test/main/promise.js

```diff
@@ -247,4 +265,25 @@ QUnit.module('Support for Promise', function () {
 
     return createMockPromise(assert, true, new Error('this is an error'));
   });
+
+  QUnit.module('compatible with fake timers', {
+    beforeEach: function (assert) {
+      this.setTimeout = globalNS.setTimeout;
+      globalNS.setTimeout = function () {};
+      this.setImmediate = globalNS.setImmediate;
+      globalNS.setImmediate = function () {};
+      // Adds 1 assertion
+      return createMockPromise(assert);
+    },
+    afterEach: function (assert) {
+      globalNS.setTimeout = this.setTimeout;
+      globalNS.setImmediate = this.setImmediate;
```

Comment:
This leaves behind an undefined global property in environments where the property did not exist. Can we avoid that?